src/io.c

```diff
@@ -1777,17 +1782,37 @@ _dispatch_stream_cleanup_operations(dispatch_stream_t stream,
 	dispatch_operation_t op, tmp;
 	typeof(*stream->operations) *operations;
 	operations = &stream->operations[DISPATCH_IO_RANDOM];
+#ifdef __LINUX_PORT_HDD__
+	LINUX_PORT_ERROR();
+        // NOT CORRECT (doesn't support removal during iteration)
+	TAILQ_FOREACH(op, operations, operation_list) {
```

Comment:
Agreed.  This was intended as a really short term stop-gap to get things to build.  Not something that should be merged back.  We'll at least pull in the missing safe macros to shims.